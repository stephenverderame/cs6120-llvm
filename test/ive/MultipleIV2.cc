// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    int k = 100;
    for (int i = 5000; i >= m; i -= m) {
        k -= m;
        printf("%d\n", i);
        printf("%d\n", k);
    }
}
// NOLINTEND