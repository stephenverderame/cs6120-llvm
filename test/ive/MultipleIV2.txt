5000
80
4980
60
4960
40
4940
20
4920
0
4900
-20
4880
-40
4860
-60
4840
-80
4820
-100
4800
-120
4780
-140
4760
-160
4740
-180
4720
-200
4700
-220
4680
-240
4660
-260
4640
-280
4620
-300
4600
-320
4580
-340
4560
-360
4540
-380
4520
-400
4500
-420
4480
-440
4460
-460
4440
-480
4420
-500
4400
-520
4380
-540
4360
-560
4340
-580
4320
-600
4300
-620
4280
-640
4260
-660
4240
-680
4220
-700
4200
-720
4180
-740
4160
-760
4140
-780
4120
-800
4100
-820
4080
-840
4060
-860
4040
-880
4020
-900
4000
-920
3980
-940
3960
-960
3940
-980
3920
-1000
3900
-1020
3880
-1040
3860
-1060
3840
-1080
3820
-1100
3800
-1120
3780
-1140
3760
-1160
3740
-1180
3720
-1200
3700
-1220
3680
-1240
3660
-1260
3640
-1280
3620
-1300
3600
-1320
3580
-1340
3560
-1360
3540
-1380
3520
-1400
3500
-1420
3480
-1440
3460
-1460
3440
-1480
3420
-1500
3400
-1520
3380
-1540
3360
-1560
3340
-1580
3320
-1600
3300
-1620
3280
-1640
3260
-1660
3240
-1680
3220
-1700
3200
-1720
3180
-1740
3160
-1760
3140
-1780
3120
-1800
3100
-1820
3080
-1840
3060
-1860
3040
-1880
3020
-1900
3000
-1920
2980
-1940
2960
-1960
2940
-1980
2920
-2000
2900
-2020
2880
-2040
2860
-2060
2840
-2080
2820
-2100
2800
-2120
2780
-2140
2760
-2160
2740
-2180
2720
-2200
2700
-2220
2680
-2240
2660
-2260
2640
-2280
2620
-2300
2600
-2320
2580
-2340
2560
-2360
2540
-2380
2520
-2400
2500
-2420
2480
-2440
2460
-2460
2440
-2480
2420
-2500
2400
-2520
2380
-2540
2360
-2560
2340
-2580
2320
-2600
2300
-2620
2280
-2640
2260
-2660
2240
-2680
2220
-2700
2200
-2720
2180
-2740
2160
-2760
2140
-2780
2120
-2800
2100
-2820
2080
-2840
2060
-2860
2040
-2880
2020
-2900
2000
-2920
1980
-2940
1960
-2960
1940
-2980
1920
-3000
1900
-3020
1880
-3040
1860
-3060
1840
-3080
1820
-3100
1800
-3120
1780
-3140
1760
-3160
1740
-3180
1720
-3200
1700
-3220
1680
-3240
1660
-3260
1640
-3280
1620
-3300
1600
-3320
1580
-3340
1560
-3360
1540
-3380
1520
-3400
1500
-3420
1480
-3440
1460
-3460
1440
-3480
1420
-3500
1400
-3520
1380
-3540
1360
-3560
1340
-3580
1320
-3600
1300
-3620
1280
-3640
1260
-3660
1240
-3680
1220
-3700
1200
-3720
1180
-3740
1160
-3760
1140
-3780
1120
-3800
1100
-3820
1080
-3840
1060
-3860
1040
-3880
1020
-3900
1000
-3920
980
-3940
960
-3960
940
-3980
920
-4000
900
-4020
880
-4040
860
-4060
840
-4080
820
-4100
800
-4120
780
-4140
760
-4160
740
-4180
720
-4200
700
-4220
680
-4240
660
-4260
640
-4280
620
-4300
600
-4320
580
-4340
560
-4360
540
-4380
520
-4400
500
-4420
480
-4440
460
-4460
440
-4480
420
-4500
400
-4520
380
-4540
360
-4560
340
-4580
320
-4600
300
-4620
280
-4640
260
-4660
240
-4680
220
-4700
200
-4720
180
-4740
160
-4760
140
-4780
120
-4800
100
-4820
80
-4840
60
-4860
40
-4880
20
-4900
