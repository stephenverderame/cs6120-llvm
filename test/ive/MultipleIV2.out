Loop: 
1:                                                ; preds = %8, %0
  %.01 = phi i32 [ 100, %0 ], [ %5, %8 ]
  %.0 = phi i32 [ 5000, %0 ], [ %9, %8 ]
  %2 = icmp sge i32 %.0, 20
  br i1 %2, label %4, label %3

Basic IV:   %5 = sub nsw i32 %.01, 20
Basic IV:   %9 = sub nsw i32 %.0, 20
Basic IV:   %.0 = phi i32 [ 5000, %0 ], [ %9, %8 ]
Basic IV:   %.01 = phi i32 [ 100, %0 ], [ %5, %8 ]
